src/Extensions/Diagnostics.HealthChecks/Internal/RestHealthCheckPublisher.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.ObjectPool;
+using Microsoft.Extensions.Options;
+using Microsoft.Omex.Extensions.Abstractions;
+using Microsoft.ServiceFabric.Client;
+using Microsoft.ServiceFabric.Client.Http;
+using ServiceFabricCommon = Microsoft.ServiceFabric.Common;
+using ServiceFabricHealth = System.Fabric.Health;
+
+namespace Microsoft.Omex.Extensions.Diagnostics.HealthChecks
+{
+	internal class RestHealthCheckPublisher : HealthCheckPublisher
+	{
+		private readonly RestClientWrapper m_clientWrapper;
+
+		private IServiceFabricClient? m_client;
+
+		private readonly ILogger<ServiceFabricHealthCheckPublisher> m_logger;
+
+		private string? m_nodeName;
+
+		internal static new string? HealthReportSourceId => nameof(RestHealthCheckPublisher);
+
+		internal const string NodeNameVariableName = "Fabric_NodeName";
+
+		public RestHealthCheckPublisher(ILogger<ServiceFabricHealthCheckPublisher> logger,
+										IOptions<RestHealthCheckPublisherOptions> options,
+										ObjectPoolProvider objectPoolProvider) : base(objectPoolProvider)
+		{
+			m_nodeName = FindNodeName();
+			m_clientWrapper = new(new Uri(options.Value.RestHealthPublisherClusterEndpoint));
+			m_logger = logger;
+		}
+
+		public override async Task PublishAsync(HealthReport report, CancellationToken cancellationToken)
+		{
+			if(m_nodeName == null)
+			{
+				m_logger.LogError(Tag.Create(), "Can't find node name.");
+				return;
+			}
+
+			m_client = await m_clientWrapper.GetAsync();
+
+			if(m_client == null)
+			{
+				m_logger.LogInformation(Tag.Create(), "HTTP Client isn't ready yet.");
+				return;
+			}
```

Comment:
By definition, your function `GetAsync` has not nullable return type, so I don't think a null check needed here.